Firmware/MotorControl/commands.cpp

```diff
@@ -113,12 +114,40 @@ void enter_dfu_mode() {
     NVIC_SystemReset();
 }
 
+#if HW_VERSION_MAJOR == 3
+const uint8_t* otp_ptr =
+    *(uint8_t*)0x1fff7800 == 0xfe ? (uint8_t*)0x1fff7800 :
+    *(uint8_t*)0x1fff7800 != 0x00 ? NULL :
+        *(uint8_t*)0x1fff7810 == 0xfe ? (uint8_t*)0x1fff7810 : NULL;
+
+// Read hardware version from OTP if available, otherwise fall back
+// to software defined version.
+const uint8_t board_version_major = otp_ptr ? otp_ptr[3] : HW_VERSION_MAJOR;
+const uint8_t board_version_minor = otp_ptr ? otp_ptr[4] : HW_VERSION_MINOR;
+const uint8_t board_version_variant = otp_ptr ? otp_ptr[5] : (HW_VERSION_VOLTAGE == 24 ? 0 : 1);
```

Comment:
Comparing with exactly 24 is not very stable, the nominal voltage may be different in the future.
Why not just expose the voltage instead of a variant-number?